### DISCUSSION TITLE
v0.4.3

### DISCUSSION THREAD

**Discussion by FormerLurker** [FormerLurker/Octolapse#907] [General]
# Summary

This release fixes installation issues, specifically Pillow not installing, or breaking after it is manually installed.  I want to thank @SuperSandro2000, @r-stiller, and @jneilliii for submitting PRs to attempt to fix the issue with PRs.  Ultimately, there were other changes that were required, but I was put on the right track and prodded appropriately to get this fix pushed out!   

It also contains a new feature which allows @OCOTLAPSE commands while using all smart triggers, a bug fix for logging caused by removal of Python2 Six library, and some corrected text.

# Enhancements

*  @OCOTLAPSE TAKE-SNAPSHOT now supported in ALL smart triggers!  This should allow you to get a picture of your empty build plate, and a snapshot when the print is 100% completed just by adding `@OCTOLAPSE TAKE-SNAPSHOT` to your start and end gcode!  You can enable or disable this in the trigger settings (default = enabled).

# Bug Fixes

* Change pillow version to "pillow>=9.3,<11", and handle all valid imports (from pil import Image and from PIL import Image).
* Fix logging issue caused by removal of Six
* Fix typo in the 'Videos and Images' close button (previously 'Cancel') thanks to a PR by @matt-lethargic!

# Notes

I'm trying my hardest to keep this up-to-date and working for everyone, and will continue to do so to the best of my abilities.  I've had some setbacks recently that really have eaten into my free time, but am working tirelessly to get back to a better place.  Please be patient with me, thanks!

<hr /><em>This discussion was created from the release <a href='https://github.com/FormerLurker/Octolapse/releases/tag/v0.4.3'>v0.4.3</a>.</em>

**Comment by s1ckopsycho** [comment]
"I'm trying my hardest to keep this up-to-date and working for everyone" - THANK YOU!  Such a fine product, I cant imagine my printer without it.  It's so impressive even in it's current iteration, I couldn't imagine people complaining about improvements moving too slow.  Thank you for all the hard work and of course for keeping it open.
![Project-Management-Speed-Cost-Quality](https://github.com/FormerLurker/Octolapse/assets/40005697/09b35664-4fef-4811-90b3-f45365b10209)

**Comment by Siggy101** [comment]
Thank you so much @FormerLurker for all your efforts.
We, the users and community, can sometimes be a bit quiet, just consuming all of your hard work without even a 'thank you', but trust me, we are hugely grateful for your effort.  
To write code, good code, and publish it for everyone to use it for free, is a very altruistic thing.  You can be proud of what you have made.  The plugin works brilliantly and I am sure that thousands of users just count on it every day without thinking about who made it and what it took.
Keep up the epic work, it is very much appreciated!

**Reply by s1ckopsycho** [reply]
I so much agree.  As I said, I can't imagine my printer without it, and to have such humble words from the coder who works so diligently on FREE software that, honestly, I'd be more than happy to pay for.  Whenever I speak to people about 3d printing, I ALWAYS bring up octolapse and how cool it is, the many variations of options, the support for so many cameras, etc etc.  It's one thing to have a printed part, but another to see it all done in action.  I'll try to buy you a cup of coffee when I can. =)

**Reply by FormerLurker** [reply]
Thanks everyone, I really needed the pick-me-up.  You are the reason I keep on keeping on.  Thank you.  Thank you.  Thank you again.

**Comment by wstacey** [comment]
Is there a way to get more in depth explanation of how to use the @OCTOLAPSE commands? 

I tried adding @OCOTLAPSE TAKE-SNAPSHOT to the start of the gcode in the slicer and in the starting Gcode scripts on Octoprint but it doesn't do anything. I can't seem to get Ocotprint to take a snapshot of the empty bed before the print starts.

**Reply by FormerLurker** [reply]
(posted from the wrong account earlier, sorry.  reposting)

Looks like you have a typo.  Change

```@OCOTLAPSE TAKE-SNAPSHOT```

to

```@OCTOLAPSE TAKE-SNAPSHOT```

Did I mess this up somewhere?  If so, please let me know and I'll correct it ASAP!

**Reply by wstacey** [reply]
There was a typo when typing in here, but the code i copied into the gcode scripts was copied from the release notes and is correct. Still doesn't work. 
![image](https://github.com/FormerLurker/Octolapse/assets/7769733/f01f09c9-199f-46e4-946c-770eb7aa90f7)

**Reply by FormerLurker** [reply]
Yeah, i don't think that will work currently, since Octolapse pre-processes the gcode file itself, but Octoprint injects those commands in real-time.  This means that the preprocessor will never see that command.

If you tried this in your slicer, please upload your gcode file and drop a link.  I will take a look at it.

I am sorry there is not better documentation currently, but I accelerated the rollout of this feature due to some critical fixes that were needed to resolve some installation issues.  In fact, I never actually received any feedback on this feature from the original feature request issue, so I may be the only one who was ever able to use it successfully, 😂

**Reply by wstacey** [reply]
Here is the link to the latest gcode file I tried. When printing this file it didn't grab a snapshot until after the first layer completed.

https://drive.google.com/file/d/1LFyxp9Pb1l-PHy_2b2kIYCTINDXPNca5/view?usp=share_link


Also Thank you so much for taking the time to look into this!

**Reply by FormerLurker** [reply]
Move the TAKE-SNAPSHOT command after the G28 command and see if that helps.  Who knows where your extruder would be prior to that, right!  Octolapse won't take any snapshots until it's 100% certain it knows where your hot end is.

I would also recommend adding a G0 command after the G28, so that your code looks like this:

``` 
G28 ;Home
G0 X0 Y0 Z0.2; Move to 0,0,0.2
@OCTOLAPSE TAKE-SNAPSHOT
``` 

To ensure a position is known.  Some printers home in different locations.  There is a setting in the printer profile where you can set the home position, but this is a good way to do that too.  Notice I set the Z position to 0.2mm, which is to protect your bed.  You might want to set that to the lowest layer height you ever use.

Let me know how it goes!

**Reply by wstacey** [reply]
sorry for the late response. Got caught up working on another project and hadn't had an opportunity to test this. It still doesn't work. Still only get a screenshot after the first layer is complete.

**Comment by Bock326** [comment]
Yo i appreciate you

**Comment by s1ckopsycho** [comment]
Hey @FormerLurker- quick question.  Will Octolapse support the full resolution of the Picam V3 (Horizontal/vertical: 4608 × 2592 pixels)?  I have changed the resolution in the octoprint.txt file, but I never went all the way up to its highest resolution.  Just curious, as obviously I could just try it myself and see.  I was reading more of your documentation on "Configuring a Raspberry Pi Camera" and it says to do a dist-upgrade.  Well, I've got PTSD on that since last time I did it it totally killed octoprint and I had to start from scratch.  This time I'll remember... Jesus saves, but Bhudda does incremental backups. =)

**Reply by FormerLurker** [reply]
You might be fine with the current distro.  That guide is a little on the old side.

Regarding the resolution, Octolapse will happily grab an image of any resolution.  However, there are some restrictions on the final rendered image depending on the codec you choose in the rendering profile.  I have an explanation of some of this in the context help next to the codec drop down I believe.  If you run into issues, you could always download and render the video in whatever software you choose.

**Comment by dchamb55** [comment]
I am unable to install V0.4.3. I am on V0.4.1 Here is the dialog from the failed update:

Updating, please wait.
++++++++++++++++++++++++++++++++
Now updating Octolapse to v0.4.3
++++++++++++++++++++++++++++++++
/home/pi/oprint/bin/python2 -m pip --disable-pip-version-check install https://github.com/FormerLurker/Octolapse/archive/v0.4.3.zip --no-cache-dir
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting https://github.com/FormerLurker/Octolapse/archive/v0.4.3.zip
Downloading https://github.com/FormerLurker/Octolapse/archive/v0.4.3.zip
ERROR: Could not find a version that satisfies the requirement pillow<11,>=9.3 (from Octolapse==0.4.3) (from versions: 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.7.5, 1.7.6, 1.7.7, 1.7.8, 2.0.0, 2.1.0, 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.3.2, 2.4.0, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0, 2.6.1, 2.6.2, 2.7.0, 2.8.0, 2.8.1, 2.8.2, 2.9.0, 3.0.0, 3.1.0rc1, 3.1.0, 3.1.1, 3.1.2, 3.2.0, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.4.0, 3.4.1, 3.4.2, 4.0.0, 4.1.0, 4.1.1, 4.2.0, 4.2.1, 4.3.0, 5.0.0, 5.1.0, 5.2.0, 5.3.0, 5.4.0, 5.4.1, 6.0.0, 6.1.0, 6.2.0, 6.2.1, 6.2.2)
ERROR: No matching distribution found for pillow<11,>=9.3 (from Octolapse==0.4.3)
The update did not finish successfully. Please consult <code>octoprint.log</code> and <code>plugin_softwareupdate_console.log</code> for details.
